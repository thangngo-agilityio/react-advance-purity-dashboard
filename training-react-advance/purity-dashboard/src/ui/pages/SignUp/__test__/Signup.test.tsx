import { render } from "@testing-library/react";
import '@testing-library/jest-dom';

// Components
import SignUpPage from "..";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

const mockCreateAccount = jest.fn()

jest.mock('@/lib/hooks', () => ({
  useAuthRegister: () => ({
    createAccount: mockCreateAccount,
  }),
}));

jest.mock('@/lib/stores', () => ({
  authStore: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

jest.mock('@chakra-ui/react', () => ({
  useToast: jest.fn(),
}));

const mockToast = jest.fn();
jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useToast: () => mockToast,
}));

const queryClient = new QueryClient();
describe('Sign Up page', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('Should render match with snapshot.', () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <SignUpPage />
      </QueryClientProvider>
    );

    expect(container).toMatchSnapshot();
  });
})
