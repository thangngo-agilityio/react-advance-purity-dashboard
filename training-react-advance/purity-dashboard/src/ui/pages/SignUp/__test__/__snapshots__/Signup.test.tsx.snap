// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up page Should render match with snapshot. 1`] = `
<div>
  <div
    class="chakra-stack css-r7skei"
  >
    <div
      class="chakra-stack css-17pq9m2"
    >
      <div
        class="chakra-stack css-1ulczcx"
      >
        <div
          class="chakra-stack css-90p4mk"
        >
          <h2
            class="chakra-heading css-0"
          >
            Welcome!
          </h2>
          <p
            class="chakra-text css-xi606m"
          >
            Use these awesome forms to login or create new account in your project for free.
          </p>
        </div>
        <div
          class="css-vojejw"
        >
          <div
            class="css-0"
          >
            <div
              class="css-1pe4bzm"
            >
              <h2
                class="chakra-heading css-0"
              >
                Register with
              </h2>
            </div>
            <div
              class="chakra-stack css-pwvjb9"
            >
              <button
                class="chakra-button css-4xx2wk"
                disabled=""
                type="button"
              >
                <svg
                  fill="none"
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_92_243)"
                  >
                    <path
                      d="M28.9173 16C28.9173 8.87004 23.1307 3.08337 16.0007 3.08337C8.87065 3.08337 3.08398 8.87004 3.08398 16C3.08398 22.2517 7.52732 27.4571 13.4173 28.6584V19.875H10.834V16H13.4173V12.7709C13.4173 10.278 15.4452 8.25004 17.9382 8.25004H21.1673V12.125H18.584C17.8736 12.125 17.2923 12.7063 17.2923 13.4167V16H21.1673V19.875H17.2923V28.8521C23.8152 28.2063 28.9173 22.7038 28.9173 16Z"
                      fill="#2D3748"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_92_243"
                    >
                      <rect
                        fill="white"
                        height="31"
                        transform="translate(0.5 0.5)"
                        width="31"
                      />
                    </clippath>
                  </defs>
                </svg>
              </button>
              <button
                class="chakra-button css-4xx2wk"
                disabled=""
                type="button"
              >
                <svg
                  fill="none"
                  height="32"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.6379 8.78461C19.1967 8.78461 18.165 9.94953 16.4648 9.94953C14.7217 9.94953 13.392 8.79308 11.2765 8.79308C9.20583 8.79308 6.99769 10.0573 5.59542 12.211C3.62644 15.248 3.96066 20.9678 7.14966 25.8407C8.29036 27.585 9.81372 29.5413 11.8118 29.5625H11.8481C13.5846 29.5625 14.1004 28.4254 16.4902 28.4121H16.5266C18.8806 28.4121 19.3529 29.5558 21.0821 29.5558H21.1184C23.1165 29.5346 24.7216 27.367 25.8623 25.6294C26.6833 24.3797 26.9884 23.7524 27.6181 22.3386C23.0051 20.5876 22.264 14.0479 26.8262 11.5407C25.4336 9.79695 23.4767 8.78703 21.6319 8.78703L21.6379 8.78461Z"
                    fill="#2D3748"
                  />
                  <path
                    d="M21.1008 2.43903C19.6477 2.53772 17.9524 3.46287 16.9594 4.67078C16.0585 5.76547 15.3174 7.38934 15.608 8.96416H15.7243C17.2719 8.96416 18.8558 8.03234 19.7809 6.83836C20.6722 5.7019 21.3479 4.09135 21.1008 2.43903Z"
                    fill="#2D3748"
                  />
                </svg>
              </button>
              <button
                class="chakra-button css-4xx2wk"
                disabled=""
                type="button"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M22.1794 10.3817L22.0734 9.93222H12.3028V14.0675H18.1406C17.5345 16.9457 14.722 18.4607 12.4247 18.4607C10.7531 18.4607 8.99109 17.7575 7.82484 16.6274C7.20953 16.0216 6.71974 15.3004 6.38353 14.505C6.04731 13.7097 5.87126 12.8558 5.86547 11.9924C5.86547 10.2505 6.64828 8.50816 7.78734 7.36206C8.9264 6.21597 10.6467 5.57472 12.3572 5.57472C14.3161 5.57472 15.72 6.61488 16.245 7.08925L19.1836 4.16613C18.3216 3.40863 15.9534 1.49988 12.2625 1.49988C9.41484 1.49988 6.68437 2.59066 4.68844 4.58003C2.71875 6.53894 1.69922 9.37159 1.69922 11.9999C1.69922 14.6282 2.66391 17.3192 4.57266 19.2936C6.61219 21.3992 9.50062 22.4999 12.4748 22.4999C15.1809 22.4999 17.7459 21.4396 19.5741 19.5158C21.3712 17.6221 22.3008 15.0017 22.3008 12.2549C22.3008 11.0985 22.1845 10.4117 22.1794 10.3817Z"
                    fill="#2D3748"
                  />
                </svg>
              </button>
            </div>
            <p
              class="chakra-text css-1sohl71"
            >
              Or
            </p>
          </div>
          <form
            class="chakra-stack css-1a3gvs5"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-byg028"
                for="field-:r2:"
                id="field-:r2:-label"
              >
                Name
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <input
                  aria-label="name"
                  class="chakra-input css-rak02u"
                  id="field-:r2:"
                  placeholder="name"
                  type="text"
                />
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-byg028"
                for="field-:r3:"
                id="field-:r3:-label"
              >
                Email
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <input
                  class="chakra-input css-rak02u"
                  id="field-:r3:"
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-byg028"
                for="field-:r4:"
                id="field-:r4:-label"
              >
                Password
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <input
                  class="chakra-input css-rak02u"
                  id="field-:r4:"
                  name="password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <button
                  aria-label="The eye icon"
                  class="chakra-input__right-element css-34z6oi"
                  data-testid="right-icon-input"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-1l546ul"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                      />
                      <path
                        d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-byg028"
                for="field-:r5:"
                id="field-:r5:-label"
              >
                Confirm password
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <input
                  class="chakra-input css-rak02u"
                  id="field-:r5:"
                  name="confirmPassword"
                  placeholder="Confirm password"
                  type="password"
                  value=""
                />
                <button
                  aria-label="The eye icon"
                  class="chakra-input__right-element css-34z6oi"
                  data-testid="right-icon-input"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-1l546ul"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                      />
                      <path
                        d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                      />
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </form>
          <div
            class="css-omcqsq"
          >
            <p
              class="chakra-text css-dz8804"
            />
            <button
              aria-label="Sign Up"
              class="chakra-button css-11w87qx"
              form="register-form"
              role="button"
              type="submit"
            >
              SIGN UP
            </button>
          </div>
          <div
            class="css-1l4w6pd"
          >
            <p
              class="chakra-text css-25kmip"
            >
              Already have an account?
            </p>
            <button
              aria-label="sign in"
              class="chakra-button css-iiwber"
              type="button"
            >
              Sign in
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-1ew6dnr"
      >
        <div
          class="chakra-stack css-1g9houa"
        >
          <div
            class="chakra-stack css-9hjtr8"
          >
            <p
              class="chakra-text css-0"
              content="@ 2021, Made with ❤️ by"
            />
            <p
              class="chakra-text css-yeh2r3"
              content="Creative Tim"
            />
            <p
              class="chakra-text css-0"
              content="&"
            />
            <p
              class="chakra-text css-yeh2r3"
              content="Simple"
            />
            <p
              class="chakra-text css-0"
              content="for a better web"
            />
          </div>
          <ul
            class="css-yjzj4h"
            role="list"
          >
            <li
              class="css-0"
            >
              <a
                class="chakra-link css-old1by"
                href="/creative-tim"
              >
                Creative Tim
              </a>
            </li>
            <li
              class="css-0"
            >
              <a
                class="chakra-link css-old1by"
                href="/simple"
              >
                Simple
              </a>
            </li>
            <li
              class="css-0"
            >
              <a
                class="chakra-link css-old1by"
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="css-0"
            >
              <a
                class="chakra-link css-old1by"
                href="/license"
              >
                License
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="css-ps1yoi"
    />
  </div>
</div>
`;
